fsm.go

```diff
@@ -423,6 +451,7 @@ const (
 	callbackBeforeEvent
 	callbackLeaveState
 	callbackEnterState
+	callbackOnState
```

Comment:
Maybe `BeforeState` would be a more descriptive name for this callback and related functionality?